[BUG]: Destroy of github_membership fails if the invitation in failed state

### Expected Behavior

By documentation, destroy of github_membership should cancel the invitation or remove the user.  This should included Failed Invitations as well.  

### Actual Behavior

Destroy of github_membership only cancels active invitations.  Destroy on a github_membership in a state of failed invitation causes Github API 404 and failure of apply.


Terraform will perform the following actions:

  # github_membership.all_org_members["testuser"] will be destroyed
  # (because key ["testuser"] is not in for_each map)
  - resource "github_membership" "all_org_members" {
      - downgrade_on_destroy = false -> null
      - etag                 = "W/\"74a1af95e5838e42b9d9cb2fd68ac6sdfsdf52cfc89b36179358c986c29614ab6\"" -> null
      - id                   = "repo:testuser" -> null
      - role                 = "member" -> null
      - username             = "testuser" -> null
    }

  # github_team_membership.all_org_members["testuser"] will be destroyed
  # (because key ["testuser"] is not in for_each map)
  - resource "github_team_membership" "all_org_members" {
      - etag     = "W/\"b5e8cb74d2ac42b00decd0dsdfsdf6124eac975bdf1562308f720c4716c377004\"" -> null
      - id       = "13037747:testuser" -> null
      - role     = "member" -> null
      - team_id  = "13037747" -> null
      - username = "testuser" -> null
    }

Plan: 0 to add, 0 to change, 2 to destroy.
github_team_membership.all_org_members["testuser"]: Destroying... [id=13037747:testuser]
github_team_membership.all_org_members["testuser"]: Destruction complete after 1s
github_membership.all_org_members["testuser"]: Destroying... [id=repo:testuser]
╷
│ Error: DELETE https://api.github.com/orgs/repo/memberships/testuser: 404 Cannot find testuser []
│ 
│ 
╵
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.

### Terraform Version

- Installed integrations/github v6.6.0 (signed by a HashiCorp partner, key ID 38027F80D7FD5FB2)

### Affected Resource(s)

- github_membership

### Terraform Configuration Files

```hcl

```

### Steps to Reproduce

_No response_

### Debug Output

```shell

```

### Panic Output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/github/resource_github_membership.go
+++ b/github/resource_github_membership.go
@@ -66,7 +66,7 @@ func resourceGithubMembershipCreateOrUpdate(ctx context.Context, d *schema.Resou
 		ctx = context.WithValue(ctx, ctxId, d.Id())
 	}
 
-	_, _, err = client.Organizations.EditOrgMembership(ctx,
+	_, resp, err := client.Organizations.EditOrgMembership(ctx,
 		username,
 		orgName,
 		&github.Membership{
@@ -79,7 +79,11 @@ func resourceGithubMembershipCreateOrUpdate(ctx context.Context, d *schema.Resou
 
 	d.SetId(buildTwoPartID(orgName, username))
 
-	return resourceGithubMembershipRead(ctx, d, meta)
+	if err = d.Set("etag", resp.Header.Get("ETag")); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
 }
 
 func resourceGithubMembershipRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {

--- a/github/resource_github_membership.go
+++ b/github/resource_github_membership.go
@@ -174,6 +174,17 @@ func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData,
 	} else {
 		log.Printf("[INFO] Revoking '%s' membership for '%s'", orgName, username)
 		_, err = client.Organizations.RemoveOrgMembership(ctx, username, orgName)
+		if err != nil {
+			var ghErr *github.ErrorResponse
+			if errors.As(err, &ghErr) {
+				if ghErr.Response.StatusCode == http.StatusNotFound {
+					log.Printf("[INFO] Not removing '%s' membership for '%s' because they are not a member of the org anymore", orgName, username)
+					return nil
+				}
+			}
+
+			return diag.FromErr(err)
+		}
 	}
 
 	return diag.FromErr(err)

--- a/github/resource_github_membership.go
+++ b/github/resource_github_membership.go
@@ -3,10 +3,11 @@ package github
 import (
 	"context"
 	"errors"
-	"log"
+	"fmt"
 	"net/http"
 
 	"github.com/google/go-github/v81/github"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
@@ -108,8 +109,9 @@ func resourceGithubMembershipRead(ctx context.Context, d *schema.ResourceData, m
 				return nil
 			}
 			if ghErr.Response.StatusCode == http.StatusNotFound {
-				log.Printf("[INFO] Removing membership %s from state because it no longer exists in GitHub",
-					d.Id())
+				tflog.Info(ctx, fmt.Sprintf("Removing membership %s from state because it no longer exists in GitHub", d.Id()), map[string]any{
+					"membership_id": d.Id(),
+				})
 				d.SetId("")
 				return nil
 			}
@@ -145,7 +147,11 @@ func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData,
 	downgradeTo := "member"
 
 	if downgradeOnDestroy {
-		log.Printf("[INFO] Downgrading '%s' membership for '%s' to '%s'", orgName, username, downgradeTo)
+		tflog.Info(ctx, fmt.Sprintf("Downgrading '%s' membership for '%s' to '%s'", orgName, username, downgradeTo), map[string]any{
+			"org_name": orgName,
+			"username": username,
+			"role":     downgradeTo,
+		})
 
 		// Check to make sure this member still has access to the organization before downgrading.
 		// If we don't do this, the member would just be re-added to the organization.
@@ -155,7 +161,10 @@ func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData,
 			var ghErr *github.ErrorResponse
 			if errors.As(err, &ghErr) {
 				if ghErr.Response.StatusCode == http.StatusNotFound {
-					log.Printf("[INFO] Not downgrading '%s' membership for '%s' because they are not a member of the org anymore", orgName, username)
+					tflog.Info(ctx, fmt.Sprintf("Not downgrading '%s' membership for '%s' because they are not a member of the org anymore", orgName, username), map[string]any{
+						"org_name": orgName,
+						"username": username,
+					})
 					return nil
 				}
 			}
@@ -164,7 +173,11 @@ func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData,
 		}
 
 		if *membership.Role == downgradeTo {
-			log.Printf("[INFO] Not downgrading '%s' membership for '%s' because they are already '%s'", orgName, username, downgradeTo)
+			tflog.Info(ctx, fmt.Sprintf("Not downgrading '%s' membership for '%s' because they are already '%s'", orgName, username, downgradeTo), map[string]any{
+				"org_name": orgName,
+				"username": username,
+				"role":     downgradeTo,
+			})
 			return nil
 		}
 
@@ -172,13 +185,19 @@ func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData,
 			Role: github.Ptr(downgradeTo),
 		})
 	} else {
-		log.Printf("[INFO] Revoking '%s' membership for '%s'", orgName, username)
+		tflog.Info(ctx, fmt.Sprintf("Revoking '%s' membership for '%s'", orgName, username), map[string]any{
+			"org_name": orgName,
+			"username": username,
+		})
 		_, err = client.Organizations.RemoveOrgMembership(ctx, username, orgName)
 		if err != nil {
 			var ghErr *github.ErrorResponse
 			if errors.As(err, &ghErr) {
 				if ghErr.Response.StatusCode == http.StatusNotFound {
-					log.Printf("[INFO] Not removing '%s' membership for '%s' because they are not a member of the org anymore", orgName, username)
+					tflog.Info(ctx, fmt.Sprintf("Not removing '%s' membership for '%s' because they are not a member of the org anymore", orgName, username), map[string]any{
+						"org_name": orgName,
+						"username": username,
+					})
 					return nil
 				}
 			}

--- a/github/resource_github_membership.go
+++ b/github/resource_github_membership.go
@@ -7,15 +7,16 @@ import (
 	"net/http"
 
 	"github.com/google/go-github/v81/github"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 func resourceGithubMembership() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceGithubMembershipCreateOrUpdate,
-		Read:   resourceGithubMembershipRead,
-		Update: resourceGithubMembershipCreateOrUpdate,
-		Delete: resourceGithubMembershipDelete,
+		CreateContext: resourceGithubMembershipCreateOrUpdate,
+		ReadContext:   resourceGithubMembershipRead,
+		UpdateContext: resourceGithubMembershipCreateOrUpdate,
+		DeleteContext: resourceGithubMembershipDelete,
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
@@ -49,10 +50,10 @@ func resourceGithubMembership() *schema.Resource {
 	}
 }
 
-func resourceGithubMembershipCreateOrUpdate(d *schema.ResourceData, meta any) error {
+func resourceGithubMembershipCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	err := checkOrganization(meta)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	client := meta.(*Owner).v3client
@@ -60,7 +61,6 @@ func resourceGithubMembershipCreateOrUpdate(d *schema.ResourceData, meta any) er
 	orgName := meta.(*Owner).name
 	username := d.Get("username").(string)
 	roleName := d.Get("role").(string)
-	ctx := context.Background()
 	if !d.IsNewResource() {
 		ctx = context.WithValue(ctx, ctxId, d.Id())
 	}
@@ -73,18 +73,18 @@ func resourceGithubMembershipCreateOrUpdate(d *schema.ResourceData, meta any) er
 		},
 	)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	d.SetId(buildTwoPartID(orgName, username))
 
-	return resourceGithubMembershipRead(d, meta)
+	return resourceGithubMembershipRead(ctx, d, meta)
 }
 
-func resourceGithubMembershipRead(d *schema.ResourceData, meta any) error {
+func resourceGithubMembershipRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	err := checkOrganization(meta)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	client := meta.(*Owner).v3client
@@ -92,9 +92,9 @@ func resourceGithubMembershipRead(d *schema.ResourceData, meta any) error {
 	orgName := meta.(*Owner).name
 	_, username, err := parseTwoPartID(d.Id(), "organization", "username")
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	ctx = context.WithValue(ctx, ctxId, d.Id())
 	if !d.IsNewResource() {
 		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
 	}
@@ -114,31 +114,31 @@ func resourceGithubMembershipRead(d *schema.ResourceData, meta any) error {
 				return nil
 			}
 		}
-		return err
+		return diag.FromErr(err)
 	}
 
 	if err = d.Set("etag", resp.Header.Get("ETag")); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 	if err = d.Set("username", username); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 	if err = d.Set("role", membership.GetRole()); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	return nil
 }
 
-func resourceGithubMembershipDelete(d *schema.ResourceData, meta any) error {
+func resourceGithubMembershipDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	err := checkOrganization(meta)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	client := meta.(*Owner).v3client
 	orgName := meta.(*Owner).name
-	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	ctx = context.WithValue(ctx, ctxId, d.Id())
 
 	username := d.Get("username").(string)
 	downgradeOnDestroy := d.Get("downgrade_on_destroy").(bool)
@@ -160,7 +160,7 @@ func resourceGithubMembershipDelete(d *schema.ResourceData, meta any) error {
 				}
 			}
 
-			return err
+			return diag.FromErr(err)
 		}
 
 		if *membership.Role == downgradeTo {
@@ -176,5 +176,5 @@ func resourceGithubMembershipDelete(d *schema.ResourceData, meta any) error {
 		_, err = client.Organizations.RemoveOrgMembership(ctx, username, orgName)
 	}
 
-	return err
+	return diag.FromErr(err)
 }

--- a/go.mod
+++ b/go.mod
@@ -7,6 +7,7 @@ require (
 	github.com/google/go-github/v81 v81.0.0
 	github.com/google/uuid v1.6.0
 	github.com/hashicorp/go-cty v1.5.0
+	github.com/hashicorp/terraform-plugin-log v0.9.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.38.1
 	github.com/shurcooL/githubv4 v0.0.0-20221126192849-0b5c4c7994eb
 	github.com/stretchr/testify v1.11.1
@@ -39,7 +40,6 @@ require (
 	github.com/hashicorp/terraform-exec v0.23.1 // indirect
 	github.com/hashicorp/terraform-json v0.27.1 // indirect
 	github.com/hashicorp/terraform-plugin-go v0.29.0 // indirect
-	github.com/hashicorp/terraform-plugin-log v0.9.0 // indirect
 	github.com/hashicorp/terraform-registry-address v0.4.0 // indirect
 	github.com/hashicorp/terraform-svchost v0.1.1 // indirect
 	github.com/hashicorp/yamux v0.1.2 // indirect

